File: Tycodes-brain/scraper.py
import os
from firecrawl import FirecrawlApp
from dotenv import load_dotenv

load_dotenv()

app = FirecrawlApp(api_key=os.getenv("FIRECRAWL_API_KEY"))

def get_website_markdown(url: str):
    print(f"--- Attempting to scrape: {url} ---")
    try:
        # Use .scrape() since your library doesn't recognize .scrape_url()
        result = app.scrape(url)
        
        # Access using dot notation because 'result' is a Document object
        if hasattr(result, 'markdown') and result.markdown:
            print(f"--- Scrape Successful for {url} ---")
            return result.markdown
        
        # Fallback if it's a dictionary or object without .markdown
        return str(result)
        
    except Exception as e:
        print(f"Scrape failed: {str(e)}")
        return f"Error: {str(e)}"

File: Tycodes-brain/src/scraper.ts
export async function getWebsiteMarkdown(url: string, apiKey: string): Promise<string> {
    if (!url.startsWith("http://") && !url.startsWith("https://")) {
        url = `https://${url}`;
    }

    console.log(`--- Attempting to scrape: ${url} ---`);
    if (!apiKey) {
        return "Error: FIRECRAWL_API_KEY not set";
    }

    try {
        const response = await fetch("https://api.firecrawl.dev/v1/scrape", {
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                "Authorization": `Bearer ${apiKey}`
            },
            body: JSON.stringify({
                url: url,
                formats: ["markdown"]
            })
        });

        if (!response.ok) {
            throw new Error(`HTTP error! status: ${response.status}`);
        }

        const result: any = await response.json();
        
        if (result.success && result.data && result.data.markdown) {
            console.log(`--- Scrape Successful for ${url} ---`);
            return result.data.markdown;
        }
        
        return JSON.stringify(result);
    } catch (error: any) {
        console.error(`Scrape failed: ${error.message}`);
        return `Error: ${error.message}`;
    }
}
